packages/loot-design/src/components/modals/NordigenExternalMsg.js

```diff
@@ -0,0 +1,180 @@
+  Button,
+  P,
+  ModalButtons,
+  Strong
+} from '../common';
+
+function renderError(error) {
+  return (
+    <Error style={{ alignSelf: 'center' }}>
+      {error === 'timeout'
+        ? 'Timed out. Please try again.'
+        : 'An error occurred while linking your account, sorry!'}
+    </Error>
+  );
+}
+
+export default function NordigenExternalMsg({
+  modalProps,
+  onMoveExternal,
+  onSuccess,
+  onClose: originalOnClose
+}) {
+  let [waiting, setWaiting] = useState(null);
+  let [success, setSuccess] = useState(false);
+  let [institutionId, setInstitutionId] = useState('default');
+  let [error, setError] = useState(null);
+  let data = useRef(null);
+
+  async function onJump() {
+    setError(null);
+    setWaiting('browser');
+
+    let res = await onMoveExternal({ institutionId });
+    if (res.error) {
+      setError(res.error);
+      setWaiting(null);
+      return;
+    }
+
+    data.current = res.data;
+    setWaiting(null);
+    setSuccess(true);
+  }
+
+  function onClose() {
+    originalOnClose && originalOnClose();
+    modalProps.onClose();
+  }
+
+  async function onContinue() {
+    setWaiting('accounts');
+    await onSuccess(data.current);
+    setWaiting(null);
+  }
+
+  const renderLinkButton = () => {
+    return (
+      <View>
+        <Strong>Choose your banks:</Strong>
+        <CustomSelect
+          options={[
+            ['default', 'Choose your bank'],
+            ['ING_PL_INGBPLPW', 'ING PL'],
+            ['MBANK_RETAIL_BREXPLPW', 'MBANK'],
+            ['SANDBOXFINANCE_SFIN0000', 'DEMO - TEST']
+          ]}
+          disabledKeys={['default']}
+          onChange={val => {
+            setInstitutionId(val);
+          }}
+          value={institutionId || 'default'}
+        />
+        <Button
+          primary
+          style={{
+            padding: '10px 0',
+            fontSize: 15,
+            fontWeight: 600,
+            marginTop: 10
+          }}
+          onClick={onJump}
+          disabled={institutionId === 'default'}
+        >
+          Link bank in browser &rarr;
+        </Button>
+      </View>
+    );
+  };
+
+  return (
+    <Modal
+      title="Link Your Bank"
+      {...modalProps}
+      onClose={onClose}
+      style={{ flex: 0 }}
+    >
+      {() => (
+        <View>
+          <P style={{ fontSize: 15 }}>
+            To link your bank account, you will be moved to your browser for
+            enhanced security. Click below and Actual will automatically resume
+            when you have given your bank{"'"}s credentials.
+          </P>
+
+          {error && renderError(error)}
+
+          {waiting ? (
+            <View style={{ alignItems: 'center', marginTop: 15 }}>
+              <AnimatedLoading
+                color={colors.n1}
+                style={{ width: 20, height: 20 }}
+              />
+              <View style={{ marginTop: 10, color: colors.n4 }}>
+                {waiting === 'browser'
+                  ? 'Waiting on browser...'
```

Comment:
```suggestion
                  ? 'Waiting on Nordigen...'
```


(or something else giving more detail on why we’re waiting)